src/Database/PostgreSQL/PQTypes/Model/Trigger.hs

```diff
@@ -0,0 +1,284 @@
+-- |
+-- Module: Database.PostgreSQL.PQTypes.Model.Trigger
+--
+-- Trigger name must be unique among triggers of same table. Only @CONTRAINT@ triggers are
+-- supported. They can only be run @AFTER@ an event. The associated functions are always
+-- created with no arguments and always @RETURN NULL@.
+--
+-- For details, see <https://www.postgresql.org/docs/11/sql-createtrigger.html>.
+
+module Database.PostgreSQL.PQTypes.Model.Trigger (
+  -- * Trigger functions
+    TriggerFunction(..)
+  , sqlCreateTriggerFunction
+  -- * Triggers
+  , TriggerEvent(..)
+  , Trigger(..)
+  , triggerMakeName
+  , triggerBaseName
+  , sqlCreateTrigger
+  , getDBTriggers
+  -- TODO testing; remove when PR is ready
+  -- , testDB
+  -- , testGetDBTriggers
+  -- , testTrigger1
+  -- , testTrigger2
+  ) where
+
+import Data.Bits (testBit)
+import Data.Int
+import Data.Monoid.Utils
+import Data.Set (Set)
+import Data.String
+import Database.PostgreSQL.PQTypes
+import Database.PostgreSQL.PQTypes.SQL.Builder
+import qualified Data.Set as Set
+import qualified Data.Text as Text
+
+-- | Function associated to a trigger.
+--
+-- @since 1.15.0.0
+data TriggerFunction = TriggerFunction {
+    tfName   :: RawSQL ()
+    -- ^ The function's name.
+  , tfSource :: RawSQL ()
+    -- ^ The functions's body source code.
+} deriving (Show)
+
+instance Eq TriggerFunction where
+  -- Since the functions have no arguments, it's impossible to create two functions with
+  -- the same name. Therefore comparing functions only by their names is enough in this
+  -- case. The assumption is, of course, that the database schema is only changed using
+  -- this framework.
+  f1 == f2 = tfName f1 == tfName f2
```

Comment:
It's used only in one place, but since this is a module in a public library, I think the Eq instance should be provided. Providing a way to compare two functions makes sense, in my opinion.